src/common/query-parameters.ts

```diff
@@ -0,0 +1,22 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { Transform } from 'class-transformer';
+import { IsNumber, IsOptional, IsString } from 'class-validator';
+
+export class QueryParameters {
+  @IsString()
+  @IsOptional()
+  @ApiProperty({ example: 'dumplings', type: 'string' })
+  name?: string;
+
+  @Transform(({ value }) => (value ? parseInt(value) : null))
```

Comment:
it's a good practice to always provide radix when using parseInt.
Here you can read why
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt